File: day_01/day01_a.lua
local file = 'input.txt'

local function sort_func(left, right)
    return left < right
end

-- list
local a_list = {}
local b_list = {}

for line in io.lines(file) do
    local a, b = string.match(line, '(%d+)%s+(%d+)')
    table.insert(a_list, a)
    table.insert(b_list, b)
end

for _, v in pairs(a_list) do
    print(v)
end

table.sort(a_list, sort_func)
table.sort(b_list, sort_func)

print("Sorted 📋: ")

local abs_values = {}
for i = 1, #a_list do
    local abs_val = math.abs(a_list[i] - b_list[i])
    print("abs(" .. a_list[i] .. "-" .. - b_list[i] .. "=" .. abs_val .. ")")
    table.insert(abs_values, abs_val)
end

local result = 0
for _, value in pairs(abs_values) do
    result = result + value
end

print("Result 🏁:" .. result)

File: day_01/day01_a.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	file, _ := os.Open("input.txt")
	fscanner := bufio.NewScanner(file)
	var a_list []int64
	var b_list []int64
	for fscanner.Scan() {
		pairs := strings.Split(fscanner.Text(), "   ")

		a, _ := strconv.ParseInt(pairs[0], 10, 64)
		b, _ := strconv.ParseInt(pairs[1], 10, 64)
		a_list = append(a_list, a)
		b_list = append(b_list, b)
	}

	// Sort lists
	sort.Slice(a_list, func(i, j int) bool {
		return a_list[i] < a_list[j]
	})
	sort.Slice(b_list, func(i, j int) bool {
		return b_list[i] < b_list[j]
	})

	fmt.Println(a_list)
	fmt.Println(b_list)

	var result int64 = 0
	for i := range a_list {
		tmp := int64(math.Abs(float64(a_list[i]) - float64(b_list[i])))
		fmt.Printf("Value of math.Abs: %d \n", tmp)
		result = result + tmp
	}
	print(result)

}
